tests/by-util/test_truncate.rs

```diff
@@ -420,3 +420,27 @@ fn test_fifo_error_reference_and_size() {
         .no_stdout()
         .stderr_contains("cannot open 'fifo' for writing: No such device or address");
 }
+
+#[test]
+#[cfg(target_os = "linux")]
+fn test_truncate_non_utf8_paths() {
+    use std::fs;
+
+    let ts = TestScenario::new(util_name!());
+    let at = &ts.fixtures;
+
+    // Create test file with normal name first
+    at.write("temp.txt", "test content");
+
+    // Rename to non-UTF-8 name
+    #[cfg(unix)]
+    {
+        use std::os::unix::ffi::OsStrExt;
+        let file_name = std::ffi::OsStr::from_bytes(b"test_\xFF\xFE.txt");
+
+        fs::rename(at.subdir.join("temp.txt"), at.subdir.join(file_name)).unwrap();
```

Comment:
What's the reason for renaming the file? Why not create the file in a similar way as it is done in `test_tsort.rs`?